Lib/fontTools/pens/areaPen.py

```diff
@@ -0,0 +1,57 @@
+"""Calculate the area of a glyph."""
+
+import math
+from fontTools.pens.basePen import BasePen
+from collections import namedtuple
+
+
+P = namedtuple('P', ('x', 'y'))
+
+
+def distance(p0, p1):
+    return math.hypot(p0.x - p1.x, p0.y - p1.y)
+
+
+def interpolate(p0, p1, t):
+    return P(p0.x * (1 - t) + p1.x * t, p0.y * (1 - t) + p1.y * t)
+
+
+def polygon_area(p0, p1):
+    return (p1.x - p0.x) * (p1.y + p0.y) * 0.5
+
+
+def quadratic_curve_area(p0, p1, p2):
```

Comment:
This seems to be unused.  Can you hook it up to the pen?

Also, I can give you direct code calculating this, instead of up-ordering.